core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -121,8 +125,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           context.reply(RegisterExecutorFailed("Duplicate executor ID: " + executorId))
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
-          context.reply(RegisteredExecutor)
-
+          context.reply(RegisteredExecutor(latestTokens))
```

Comment:
so if an executor dies, and by this I assume the entire container dies, the AM would relaunch a new container and when it does this I would have expected it to send the new credentials it acquired by doing the loginfromkeytab. See the ExecutorRunnable.startContainer where it writes the credentials into the launch context.

Is that not happening?  Or is there a case I'm not thinking of?